clustering/clustering.py

```diff
@@ -0,0 +1,446 @@
+import os, sys
+from time import time
+import logging
+from django.conf import settings
+
+import clustering_settings as clust_settings
+
+# The following packages are only needed if the running process is configured to be a Celery worker. 
+# We avoid importing them in appservers to avoid having to install unneeded dependencies.
+if settings.IS_CELERY_WORKER:
+    from gaia_wrapper import GaiaWrapperClustering
+    import numpy as np
+    from sklearn import metrics
+    from sklearn.feature_selection import mutual_info_classif
+    import json
```

Comment:
things from the standard library can be in the main import block. Again, order these alphabetically to make it easier to follow